getconf/base.py

```diff
@@ -23,7 +23,37 @@
     logger.addHandler(compat.NullHandler())
 
 
-ConfigKey = collections.namedtuple('ConfigKey', ['section', 'entry', 'envvar', 'doc'])
+_ConfigKey = collections.namedtuple(
+    'ConfigKey',
+    ['section', 'entry', 'envvar', 'doc', 'default', 'get_type']
+)
+
+
+class ConfigKey(_ConfigKey):
+
+    def __hash__(self):
+        # Do not hash based on default
+        return hash((self.section, self.entry, self.envvar, self.doc, self.get_type))
+
+    def to_INI(self):
+        # Remove any new line in self.doc
+        one_line_doc = ' '.join(self.doc.split())
+        parts = ['; type=%s' % self.get_type]
+        if one_line_doc:
+            parts.append(' - %s\n' % one_line_doc)
+        else:
+            parts.append('\n')
+        if self.get_type == 'list':
+            parts.append(';%s =\n' % self.entry)
+            if self.default:
+                parts.append(';  ')
+                parts.append('\n;  '.join(self.default))
+                parts.append('\n')
+        else:
+            # Make sure to have 0 or 1 for bool
+            default = int(self.default) if self.get_type == 'bool' else self.default
```

Comment:
We support much more values for ``bool``, using ``on/off`` or ``true/false`` would be closer to the real-world use.